tests/test_deserialization.py

```diff
@@ -493,6 +494,15 @@ def test_function_field_deserialization_with_callable(self):
                                 deserialize=lambda val: val.upper())
         assert field.deserialize('foo') == 'FOO'
 
+    def test_function_field_deserialization_with_context(self):
+        field = fields.Function(
+            lambda x: None,
+            deserialize=lambda val, context: val.upper() + context['key']
+        )
+        field.parent = mock.MagicMock()
```

Comment:
I don't think there's a need to use a `Mock` here. `parent` will be either another field or a Schema, so you can set `field.parent` to a Schema instance with context.